### DISCUSSION TITLE
Is there a way to blur a specific widget on Wayland?

### DISCUSSION THREAD

**Discussion by colin-heffernan** [elkowar/eww#586] [Q&A] [answered]
I want to make a blur background for my start menu, but I don't know how to target a specific widget. I'm on Hyprland.
EDIT: The only way I know how to blur is by blurring the entire LayerSurface, but that makes it so that the rounded corners of the widgets also blur.

**Comment by viandoxdev** [comment, marked as answer]
Not really possible AFAIK blurring is something only the compositor can properly implement, I think pretty much anything else (outside of an eventual extension) is a hack. The only way I can think of doing that would be to save what is initially in the buffer the compositor gives us and manually blur that. But that's hacky as hell, probably broken and way past what GTK allows (let alone eww)

**Reply by colin-heffernan** [reply]
Got it. I guess I'll just go without for the time being.

**Comment by mewoocat** [comment]
Could you elaborate on blurring the entire LayerSurface?  How is this accomplished?

**Reply by charlesrocket** [reply]
@mewoocat with `namespace` on the window. Then you just blur that layer (in hyprland/etc). It is not released yet so you need to build from source.
